### PR TITLE
3515 1

### PR DESCRIPTION
**Pull request by showthreadx**
## Номер
3515

## Номер задания
1

## Ссылка на видео с демо
https://www.youtube.com/watch?v=34hUWegVJvk

## Комментарии
Хотел сделать то-то, но не успел и не понимаю.

### CODE REVIEW

File: 3515/1/top_gems.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  def gems_arrr
+    @gems_arr = GemsLoader.new
+    @gems_table_arr = []
+    @threads = []
+  end
+end
+
+a = GemsMain.new
+a.gems_main
```
Comment (andreydro) [inline 3515/1/top_gems.rb]: What do you do by calling that method? It should be clear from method call what it does. And be attentive to correct variables naming

File: 3515/1/files/url_loader.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class GemsUrlLoader
+  attr_reader :url
+
+  def initialize(gem_name)
+    @gem_for_search = gem_name
+    @url = source_code_uri
+  end
+
+  def source_code_uri
+    if check_url(@url = Gems.info(@gem_for_search)['source_code_uri'])
```
[the remark this answers is not in the document]
Comment (showthreadx) [reply 3515/1/files/url_loader.rb]: No, this class comes from gem with name 'gems'

File: 3515/1/files/gems_main.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class GemsMain
```
Comment (andreydro) [inline 3515/1/files/gems_main.rb]: Try to name classes in way than reader can get logic what that class does. From `GemsMain` I don't understand what the job of that class

File: 3515/1/files/gems_main.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    main_process
+    threads.each(&:join)
+  end
+
+  def param_getter(gem_url, gem_name)
+    gem_params = GemParamGetter.new(gem_url, gem_name)
+    gems_table_arr << gem_params.params
+  end
+
+  def table_builder(len = gems_arr.length)
```
Comment (andreydro) [inline 3515/1/files/gems_main.rb]: in default params used nil, different brackets, strings and etc. BUT not class of another method

File: 3515/1/files/gems_get_params.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  def parse_forks
+    html_page(gem_url).css('ul.pagehead-actions a')[5].text.delete('^0-9').to_i
+  end
+
+  def parse_issues
+    html_page(gem_url).css('nav.hx_reponav span')[4].text.to_i
+  end
+
+  def parse_contributors
+    @html_page = nil
```
Comment (andreydro) [inline 3515/1/files/gems_get_params.rb]: If you get some incorrect data in instance variable and you don't need that data, that means that use it not on purpose and probably you need to separate logic that writes wrong data in it. Or make not instance variable, but just method

File: 3515/1/top_gems.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+require './files/gems_main'
+
+if ARGV.empty?
+  a = WorkWithGems.new
+  a.start
+  a.terminal_table_builder
+else
+  OptionParser.new do |opts|
+    opts.on('--file[=FILE]') do |file|
+      a = WorkWithGems.new
```
Comment (andreydro) [inline 3515/1/top_gems.rb]: Why do you still have `a` variable? Name it better

**Pull request merged**